tdvt/tdvt/config_gen/test_creator.py

```diff
@@ -0,0 +1,163 @@
+                if user_cols_dict[user_col] == TEST_ARGUMENT_DATA_TYPES[test_col]:
+                    test_args_dict[test_col].append(user_col)
+        return test_args_dict
+
+    def rewrite_tests_to_use_user_cols(self, test_sets_to_run: List[str]) -> None:
+        test_args_dict = self.map_user_cols_to_test_args()
+
+        test_suite_names = tuple('setup.' + key for key in test_sets_to_run)
+        if len(test_suite_names) == 0:
+            print("No test sets specified. Exiting.")
+            return
+
+        test_dir = get_root_dir() + '/exprtests/standard/'
+        test_setup_files = [
+            item for item in os.listdir(test_dir)
+            if item.startswith(test_suite_names)
+            and not item.startswith(CUSTOM_TABLE_EXPRESSION_TEST_EXCLUSIONS)
+        ]
+        print("Creating custom test files for the following test suites: {}".format(test_sets_to_run))
+        for test_file in test_setup_files:
+            with open(test_dir + test_file, 'r') as source_file:
+                skipped_lines = 0
+                all_lines = 0
+                lines_out = []
+                for line in source_file:
+                    all_lines += 1
+                    processed_line, skipped = self._process_line(line, test_args_dict)
+                    skipped_lines += skipped
+                    lines_out.append(processed_line)
+            output_file_name = test_file
+            if all_lines == skipped_lines:
+                output_file_name = 'SKIP.' + test_file
+            with open(self.output_dir / output_file_name, 'w') as out_file:
+                for line in lines_out:
+                    out_file.write(line)
+            print("\tCreated {}".format(self.output_dir / output_file_name))
+
+    @staticmethod
+    def _process_line(line: str, test_args_dict: List[Dict[str, List]]) -> Tuple[str, int]:
+        ignored_line = 0
+        # turn commented out tests into blank lines
+        if line.startswith('//') or line == '\n':
+            ignored_line += 1
+            return '\n', ignored_line
+        # take care of col names that are in the test args
+        else:
+            line = line.strip('\n')
+            test_col_names = TEST_ARGUMENT_DATA_TYPES.keys()
+            extant_cols = [
+                col_name for col_name in test_col_names
+                if col_name in line
+            ]
+            user_col_test_col_map = {}
+            for column in extant_cols:
+                if len(test_args_dict[column]) == 0:
+                    new_line = '// {}  {} has no matching column in the user table.\n'.format(line, column)
+                    ignored_line += 1
+                    return new_line, ignored_line
+                else:
+                    no_match = True
+                    column_at_hand = test_args_dict[column].copy()
+                    while column_at_hand and no_match:
+                        possible_user_col = random.choice(column_at_hand)
```

Comment:
I think we should prefer determinism here. Let's say I had Calcs loaded correctly, but it only differed by Column name and something else. Maybe I have to add a suffix in my system to each column and I don't support booleans. I'm thinking if I mimicked the order of the columns that map to Num in the json I could potentially achieve pretty good mapping to existing expected files..

Reply:
That makes sense and should be a common use case.

Reply:
fixed in [868d2db](https://github.com/tableau/connector-plugin-sdk/pull/1101/commits/868d2dbb7c5cfcd5b75572f792b066d9e93df153)